flexmeasures/data/schemas/scheduling/storage.py

```diff
@@ -54,10 +58,17 @@ class StorageFlexModelSchema(Schema):
     You can use StorageScheduler.deserialize_flex_config to get that filled in.
     """
 
-    soc_at_start = fields.Float(required=True, data_key="soc-at-start")
+    soc_at_start = QuantityOrValue(
+        required=True, data_key="soc-at-start", default_src_unit="MWh", to_unit="MWh"
+    )
 
-    soc_min = fields.Float(validate=validate.Range(min=0), data_key="soc-min")
-    soc_max = fields.Float(data_key="soc-max")
+    soc_min = QuantityOrValue(
+        data_key="soc-min",
+        default_src_unit="MWh",
+        to_unit="MWh",
+        validate=validate.Range(min=0),
```

Comment:
The validator is used on the deserialized value, which I believe would fail in case a string quantity definition has been deserialized to a quantity object, and is then compared to zero.